src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -552,6 +552,7 @@ btree_status_t Btree< K, V >::split_node(const BtreeNodePtr& parent_node, const
 template < typename K, typename V >
 template < typename ReqT >
 bool Btree< K, V >::is_split_needed(const BtreeNodePtr& node, const BtreeConfig& cfg, ReqT& req) const {
+    if (node->total_entries() <= 1) { return false; }
```

Comment:
It can be possible that in some key/value size setting, the total used size of a  node is more than ideal size even with one entity ( `sizeof(key[0]) + sizeof (value[0]) + record_size > ideal_size`). In this scenario, the split cannot happen.  So a check is needed that if the total entities is 1, split can be performed under no circumstance. 

If the force_split_node is enabled in one thread and the do_put returns retry, it is possible that in another thread get the lock and remove all entities in this node. So a check is needed that if the node is empty no split is needed